langchain/src/output_parsers/tests/openai_functions.int.test.ts

```diff
@@ -46,7 +48,10 @@ test("Streaming JSON patch", async () => {
   for await (const chunk of stream) {
     console.log(chunk);
     chunks.push(chunk);
-    aggregate = applyPatch(aggregate, chunk as Operation[]).newDocument;
+    aggregate = applyPatch(
+      aggregate,
+      chunk as unknown as Operation[]
```

Comment:
This seems like a regression?

Reply:
Without the cast to unknown, it gives the following error:

```ts
Conversion of type '{ setup: string; punchline: string; }' to type 'Operation[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type '{ setup: string; punchline: string; }' is missing the following properties from type 'Operation[]': length, pop, push, concat, and 29 more.ts(2352)
```

There is no overlap to the `Operation` type:

```ts
export type Operation = AddOperation<any> | RemoveOperation | ReplaceOperation<any> | MoveOperation | CopyOperation | TestOperation<any> | GetOperation<any>;

export interface BaseOperation {
    path: string;
}
export interface AddOperation<T> extends BaseOperation {
    op: "add";
    value: T;
}
export interface RemoveOperation extends BaseOperation {
    op: "remove";
}
export interface ReplaceOperation<T> extends BaseOperation {
    op: "replace";
    value: T;
}
export interface MoveOperation extends BaseOperation {
    op: "move";
    from: string;
}
export interface CopyOperation extends BaseOperation {
    op: "copy";
    from: string;
}
export interface TestOperation<T> extends BaseOperation {
    op: "test";
    value: T;
}
export interface GetOperation<T> extends BaseOperation {
    op: "_get";
    value: T;
}
export interface PatchResult<T> extends Array<OperationResult<T>> {
    newDocument: T;
}
```

But this problem should have existed before, it just wasn't reported as an error because `chunk` was an `object` but now it is strongly typed to `{ setup: string; punchline: string; }`.